client/src/utilities/contactsimporter.cpp

```diff
@@ -162,3 +182,24 @@ QVector<ContactsSet> ContactsImporter::contacts() const
 {
     return mContacts;
 }
+
+QString ContactsImporter::extractCountryFromPhoneNumber(const QString &phone) const
+{
+    if (phone.isEmpty()) {
+        return QString();
+    }
+
+    const QStringList localeName = QLocale().name().split(QStringLiteral("_"));
+    auto phoneUtil = i18n::phonenumbers::PhoneNumberUtil::GetInstance();
+    i18n::phonenumbers::PhoneNumber number;
+    if (phoneUtil->Parse(phone.toStdString(), localeName.last().toStdString(), &number) == i18n::phonenumbers::PhoneNumberUtil::NO_PARSING_ERROR) {
+        // use ZZ region to make sure that always will return country name
+        const std::string description = i18n::phonenumbers::PhoneNumberOfflineGeocoder().GetDescriptionForNumber(number,
+                                                                                                                 icu::Locale("en", "US"),
+                                                                                                                 i18n::phonenumbers::RegionCode::ZZ());
+        return QString::fromStdString(description).split(QStringLiteral(",")).last().trimmed();
```

Comment:
I find split overkill for just a lastIndexOf + mid, no point in creating and filling a temporary container. But yes I'm nitpicking, performance doesn't matter in this code, so feel free to ignore that ;)

Reply:
fixed